File: in_Java/0704 Binary Search.java
class Solution {
    public int search(int[] nums, int target) {
        int i = 0; 
        int j = nums.length-1;
        int m;
        while(i < j){
            m = i + (j-i)/2;
            if(nums[m] == target) return m;
            else if(nums[m] < target) i = m + 1;
            else j = m;
        }
        if(nums[i] == target) return i;
        return -1;
    }
}

File: in_Python/0704 Binary Search.py
class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        i = 0
        j = len(nums)-1
        while i <= j:
            m = i + (j-i) / 2
            if nums[m] == target:
                return m
            elif nums[m] > target:
                j = m - 1
            else:
                i = m + 1
        return -1